master/internal/proxy/proxy.go

```diff
@@ -156,14 +163,59 @@ func (p *Proxy) newProxyHandler(serviceID string) echo.HandlerFunc {
 		case c.IsWebSocket():
 			proxy = newSingleHostReverseWebSocketProxy(c, service.URL)
 		default:
-			proxy = httputil.NewSingleHostReverseProxy(service.URL)
+			newProxy, err := setupProxy(service.URL)
+			if err != nil {
+				return err
+			}
+
+			proxy = newProxy
 		}
 		proxy.ServeHTTP(c.Response(), req)
 
 		return nil
 	}
 }
 
+func setupProxy(serviceURL *url.URL) (*httputil.ReverseProxy, error) {
+	proxy := httputil.NewSingleHostReverseProxy(serviceURL)
+	if serviceURL.Scheme != https {
+		return proxy, nil
+	}
+	keyBytes, certBytes, err := GetMasterKeyAndCert()
+	if err != nil {
+		return proxy, err
+	}
+	cert, err := tls.X509KeyPair(certBytes, keyBytes)
+	if err != nil {
+		return proxy, nil
```

Comment:
This seems like a bug. I think you want to return the error here rather than a proxy with no error.

As an aside and a nit: I generally prefer to have functions return hard-coded zero values on error rather than an in-progress or useless thing. One of the reasons is cases like this one: a nil panic from dereferencing a nil proxy pointer after checking for a `nil` error is more likely to be caught during development.